pkg/apis/core/v1alpha1/types_seed.go

```diff
@@ -124,18 +127,26 @@ type SeedDNS struct {
 	// IngressDomain is the domain of the Seed cluster pointing to the ingress controller endpoint. It will be used
 	// to construct ingress URLs for system applications running in Shoot clusters.
 	IngressDomain string `json:"ingressDomain" protobuf:"bytes,1,opt,name=ingressDomain"`
-	// NginxIngress controls settings for a gardener managed nginx-ingress controller
+	// Provider configures a DNSProvider
 	// +optional
-	NginxIngress *SeedSettingNginxIngress `json:"nginxIngress,omitempty" protobuf:"bytes,6,opt,name=nginxIngress"`
+	Provider *SeedDNSProvider `json:"provider,omitempty" protobuf:"bytes,2,opt,name=provider`
 }
 
-type SeedSettingNginxIngress struct {
-	// Enabled controls whether a gardener managed nginx-ingress is deployed. It is used to route external traffic to the Seed.
-	Enabled bool `json:"enabled" protobuf:"bytes,1,opt,name=enabled"`
-	// Provider describes the type of the dns-provider, for example `aws-route53`, see https://github.com/gardener/external-dns-management
-	Provider string `json:"provider" protobuf:"bytes,3,opt,name=provider"`
+// SeedDNSProvider configures a DNSProvider for Seeds
+type SeedDNSProvider struct {
+	// Type describes the type of the dns-provider, for example `aws-route53`
+	Type string `json:"type" protobuf:"bytes,1,opt,name=type"`
 	// SecretRef is a reference to a Secret object containing cloud provider credentials used for registering external domains.
-	SecretRef corev1.SecretReference `json:"secretRef" protobuf:"bytes,4,opt,name=secretRef"`
+	SecretRef corev1.SecretReference `json:"secretRef" protobuf:"bytes,2,opt,name=secretRef"`
+}
+
+// IngressController configures a Gardener managed Ingress Controller listening on the ingressDomain
+type IngressController struct {
+	// Kind defines which kind of IngressController to use, for example `nginx`
+	Kind string `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+	// ProviderConfig specifies infrastructure specific configuration for the ingressController
+	// +optional
+	ProviderConfig *runtime.RawExtension `json:"providerConfig,omitempty" protobuf:"bytes,2,opt,name=providerConfig"`
```

Comment:
We should add a `domain` field here now that we have a dedicated section for the ingress controller in our API. This allows us to deprecate `.spec.dns.ingressDomain` and remove it in the next API version.

Reply:
As discussed this would make IngressController required so we did not go with it.

Reply:
Hm no, we can still use the `.spec.dns.ingressDomain` if `ingressController` is `nil`. If `ingressController.domain` is set then the `.spec.dns.ingressDomain` must not be set.

Reply:
Anyways in the long run ingressController would become required then. Because we would deprecate `spec.dns.ingressdomain`. But the IngressController is optional.

Reply:
I cannot follow, what are you suggesting?

Reply:
Right now `.spec.dns.IngressDomain` is a required field.
You suggest to
A. Add a new field `ingressDomain` under `IngressController`
B. Deprecate `dns.ingressDomain` and eventually remove it.
But if we follow through on B then `IngressController` will hold a required field and therefore needs to be required as well.
I don't think we want IngressController to become required.
Therefore, I suggest to leave `ingressDomain` under `spec.dns` and to not move it to `IngressController`

Reply:
> But if we follow through on B then IngressController will hold a required field and therefore needs to be required as well.

Hm, no, why? It's just that one of `.spec.dns.ingressDomain` OR `.spec.ingressController.domain` must be specified. This is a backwards-compatible change because
* `.spec.dns.ingressDomain` was required earlier and is now optional
* if `.spec.dns.ingressDomain` is not provided now then another field must be provided, but as `.spec.dns.ingressDomain` was required before this is okay.

Reply:
We agreed upon in a new APIVersion this would make ingressDomain required and then it would make sense to rename it to ingress. This could look like this:
If we follow along and move domain. It could look like this:
```
spec:
  ingress: // Renamed from previously IngressController
    domain: // In favor of spec.dns.ingressDomain
    controller:
      kind: // like 'nginx-ingress'
      provider-config:    
```

Reply:
ping